Crash when typing b in css file

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

1. compile zed from source for linux
2. type b in css file for autocomplete suggestion
3. crashes

https://github.com/zed-industries/zed/assets/46544691/1d39ecf6-e1ef-4afc-b00e-7033dd1fdc3f

```
Caused by:
    expected value at line 1 column 1
[2024-05-11T11:58:22-04:00 ERROR workspace::item] unexpected item event after pane was dropped
Thread "main" panicked with "called `Option::unwrap()` on a `None` value" at crates/gpui/src/platform/cosmic_text/text_system.rs:297:14
```


[full_trace.txt](https://github.com/zed-industries/zed/files/15283383/full_trace.txt)

### Environment

Zed: v1.0.0 (Zed Dev 48cba328f2235d8d2aa7126dd5d57c7e49fe13af)
OS: Linux 1.0.0
Memory: 31.3 GiB
Architecture: x86_64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<snipped log to fit in issue form>
2024-05-11T11:49:13-04:00 [INFO] Adapter "AMD Radeon RX 6950 XT (RADV NAVI21)"
2024-05-11T11:49:13-04:00 [INFO] Ray tracing is supported
2024-05-11T11:49:13-04:00 [INFO] Using surface present mode MAILBOX
2024-05-11T11:49:13-04:00 [WARN] Unable to forbid exclusive full screen
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_tile_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position_impl")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position_transformed")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_tile_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_tile_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position_impl")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position_transformed")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_tile_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_tile_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position_impl")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position_transformed")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_tile_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position_impl")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position_transformed")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_tile_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_tile_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position_impl")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position_transformed")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_tile_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position_impl")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position_transformed")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_tile_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position_impl")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position_transformed")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_tile_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_tile_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position_impl")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_device_position_transformed")
2024-05-11T11:49:13-04:00 [INFO] Skip function Some("to_tile_position")
2024-05-11T11:49:13-04:00 [INFO] set status on client 0: Authenticating
2024-05-11T11:49:13-04:00 [INFO] perform;
2024-05-11T11:49:13-04:00 [INFO] Opening main db
2024-05-11T11:49:13-04:00 [INFO] write_command; command=Auth(Some(External), Some([49, 48, 48, 48]))
2024-05-11T11:49:13-04:00 [INFO] read_command;
2024-05-11T11:49:13-04:00 [INFO] write_command; command=NegotiateUnixFD
2024-05-11T11:49:13-04:00 [INFO] read_command;
2024-05-11T11:49:13-04:00 [INFO] write_command; command=Begin
2024-05-11T11:49:13-04:00 [INFO] socket reader;
2024-05-11T11:49:13-04:00 [INFO] read_socket; self=SocketReader { socket: Async { source: Source { registration: 124, key: 0, state: Mutex { data: [Direction { tick: 0, ticks: None, waker: None, wakers: Slab { len: 0, cap: 0 } }, Direction { tick: 0, ticks: None, waker: None, wakers: Slab { len: 0, cap: 0 } }], poisoned: false, .. } }, io: Some(UnixStream { fd: FileDesc(OwnedFd { fd: 124 }), local: (unnamed), peer: "/run/user/1000/bus" (pathname) }) }, senders: Mutex { data: {Some(OwnedMatchRule(MatchRule { msg_type: Some(MethodReturn), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 0, listeners_total: 1 } }, poisoned: false, .. } }, None: Sender { inner: RwLock { data: Inner { queue: [], capacity: 64, receiver_count: 0, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event(<uninitialized>) }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Error), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 0, listeners_total: 1 } }, poisoned: false, .. } }} }, already_received_bytes: Some([]), prev_seq: 0, activity_event: Event(<uninitialized>) }
2024-05-11T11:49:13-04:00 [INFO] Opening main db
2024-05-11T11:49:13-04:00 [INFO] read_socket; self=SocketReader { socket: Async { source: Source { registration: 124, key: 0, state: Mutex { data: [Direction { tick: 0, ticks: None, waker: None, wakers: Slab { len: 0, cap: 0 } }, Direction { tick: 0, ticks: None, waker: None, wakers: Slab { len: 0, cap: 0 } }], poisoned: false, .. } }, io: Some(UnixStream { fd: FileDesc(OwnedFd { fd: 124 }), local: (unnamed), peer: "/run/user/1000/bus" (pathname) }) }, senders: Mutex { data: {Some(OwnedMatchRule(MatchRule { msg_type: Some(MethodReturn), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName("org.freedesktop.DBus"), reply-serial: 13, body: Signature("s"), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 1 } }, poisoned: false, .. } }, None: Sender { inner: RwLock { data: Inner { queue: [], capacity: 64, receiver_count: 0, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event(<uninitialized>) }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Error), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName("org.freedesktop.DBus"), reply-serial: 13, body: Signature("s"), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 1 } }, poisoned: false, .. } }} }, already_received_bytes: None, prev_seq: 1, activity_event: Event(<uninitialized>) }
2024-05-11T11:49:13-04:00 [INFO] read_socket; self=SocketReader { socket: Async { source: Source { registration: 124, key: 0, state: Mutex { data: [Direction { tick: 0, ticks: None, waker: None, wakers: Slab { len: 0, cap: 0 } }, Direction { tick: 0, ticks: None, waker: None, wakers: Slab { len: 0, cap: 0 } }], poisoned: false, .. } }, io: Some(UnixStream { fd: FileDesc(OwnedFd { fd: 124 }), local: (unnamed), peer: "/run/user/1000/bus" (pathname) }) }, senders: Mutex { data: {Some(OwnedMatchRule(MatchRule { msg_type: Some(MethodReturn), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName("org.freedesktop.DBus"), reply-serial: 13, body: Signature("s"), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 1 } }, poisoned: false, .. } }, None: Sender { inner: RwLock { data: Inner { queue: [], capacity: 64, receiver_count: 0, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event(<uninitialized>) }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Error), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName("org.freedesktop.DBus"), reply-serial: 13, body: Signature("s"), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 1 } }, poisoned: false, .. } }} }, already_received_bytes: None, prev_seq: 2, activity_event: Event(<uninitialized>) }
2024-05-11T11:49:13-04:00 [INFO] read_socket; self=SocketReader { socket: Async { source: Source { registration: 124, key: 0, state: Mutex { data: [Direction { tick: 9, ticks: Some((9, 0)), waker: None, wakers: Slab { len: 0, cap: 0 } }, Direction { tick: 0, ticks: None, waker: None, wakers: Slab { len: 0, cap: 0 } }], poisoned: false, .. } }, io: Some(UnixStream { fd: FileDesc(OwnedFd { fd: 124 }), local: (unnamed), peer: "/run/user/1000/bus" (pathname) }) }, senders: Mutex { data: {Some(OwnedMatchRule(MatchRule { msg_type: Some(MethodReturn), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 2, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 0, listeners_total: 1 } }, poisoned: false, .. } }, None: Sender { inner: RwLock { data: Inner { queue: [], capacity: 64, receiver_count: 0, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event(<uninitialized>) }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Error), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 2, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 0, listeners_total: 1 } }, poisoned: false, .. } }} }, already_received_bytes: None, prev_seq: 3, activity_event: Event(<uninitialized>) }
2024-05-11T11:49:13-04:00 [INFO] read_socket; self=SocketReader { socket: Async { source: Source { registration: 124, key: 0, state: Mutex { data: [Direction { tick: 9, ticks: Some((9, 0)), waker: None, wakers: Slab { len: 0, cap: 0 } }, Direction { tick: 0, ticks: None, waker: None, wakers: Slab { len: 0, cap: 0 } }], poisoned: false, .. } }, io: Some(UnixStream { fd: FileDesc(OwnedFd { fd: 124 }), local: (unnamed), peer: "/run/user/1000/bus" (pathname) }) }, senders: Mutex { data: {Some(OwnedMatchRule(MatchRule { msg_type: Some(MethodReturn), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName(":1.22"), reply-serial: 15, body: Signature("o"), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 2, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 1 } }, poisoned: false, .. } }, None: Sender { inner: RwLock { data: Inner { queue: [], capacity: 64, receiver_count: 0, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event(<uninitialized>) }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Error), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName(":1.22"), reply-serial: 15, body: Signature("o"), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 2, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 0 } }, poisoned: false, .. } }} }, already_received_bytes: None, prev_seq: 4, activity_event: Event(<uninitialized>) }
2024-05-11T11:49:13-04:00 [INFO] read_socket; self=SocketReader { socket: Async { source: Source { registration: 124, key: 0, state: Mutex { data: [Direction { tick: 10, ticks: Some((10, 9)), waker: None, wakers: Slab { len: 0, cap: 0 } }, Direction { tick: 0, ticks: None, waker: None, wakers: Slab { len: 0, cap: 0 } }], poisoned: false, .. } }, io: Some(UnixStream { fd: FileDesc(OwnedFd { fd: 124 }), local: (unnamed), peer: "/run/user/1000/bus" (pathname) }) }, senders: Mutex { data: {Some(OwnedMatchRule(MatchRule { msg_type: Some(MethodReturn), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 4, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 0, listeners_total: 1 } }, poisoned: false, .. } }, None: Sender { inner: RwLock { data: Inner { queue: [], capacity: 64, receiver_count: 0, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event(<uninitialized>) }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Error), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 4, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 0, listeners_total: 1 } }, poisoned: false, .. } }} }, already_received_bytes: None, prev_seq: 5, activity_event: Event(<uninitialized>) }
2024-05-11T11:49:13-04:00 [INFO] read_socket; self=SocketReader { socket: Async { source: Source { registration: 124, key: 0, state: Mutex { data: [Direction { tick: 10, ticks: Some((10, 9)), waker: None, wakers: Slab { len: 0, cap: 0 } }, Direction { tick: 0, ticks: None, waker: None, wakers: Slab { len: 0, cap: 0 } }], poisoned: false, .. } }, io: Some(UnixStream { fd: FileDesc(OwnedFd { fd: 124 }), local: (unnamed), peer: "/run/user/1000/bus" (pathname) }) }, senders: Mutex { data: {Some(OwnedMatchRule(MatchRule { msg_type: Some(MethodReturn), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 8, receiver_count: 0, inactive_receiver_count: 1, sender_count: 2, head_pos: 5, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 0 } }, poisoned: false, .. } }, None: Sender { inner: RwLock { data: Inner { queue: [], capacity: 64, receiver_count: 0, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event(<uninitialized>) }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Error), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 8, receiver_count: 0, inactive_receiver_count: 1, sender_count: 2, head_pos: 5, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 0 } }, poisoned: false, .. } }} }, already_received_bytes: None, prev_seq: 6, activity_event: Event(<uninitialized>) }
2024-05-11T11:49:13-04:00 [INFO] new;
2024-05-11T11:49:13-04:00 [INFO] {}; task_name="org.freedesktop.Secret.Item proxy caching"
2024-05-11T11:49:13-04:00 [INFO] read_socket; self=SocketReader { socket: Async { source: Source { registration: 124, key: 0, state: Mutex { data: [Direction { tick: 11, ticks: Some((11, 10)), waker: None, wakers: Slab { len: 0, cap: 0 } }, Direction { tick: 0, ticks: None, waker: None, wakers: Slab { len: 0, cap: 0 } }], poisoned: false, .. } }, io: Some(UnixStream { fd: FileDesc(OwnedFd { fd: 124 }), local: (unnamed), peer: "/run/user/1000/bus" (pathname) }) }, senders: Mutex { data: {Some(OwnedMatchRule(MatchRule { msg_type: Some(MethodReturn), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName("org.freedesktop.DBus"), reply-serial: 18, body: Signature(""), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 5, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 1 } }, poisoned: false, .. } }, None: Sender { inner: RwLock { data: Inner { queue: [], capacity: 64, receiver_count: 0, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event(<uninitialized>) }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Error), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName("org.freedesktop.DBus"), reply-serial: 18, body: Signature(""), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 5, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 1 } }, poisoned: false, .. } }} }, already_received_bytes: None, prev_seq: 7, activity_event: Event(<uninitialized>) }
2024-05-11T11:49:13-04:00 [INFO] read_socket; self=SocketReader { socket: Async { source: Source { registration: 124, key: 0, state: Mutex { data: [Direction { tick: 12, ticks: Some((12, 11)), waker: None, wakers: Slab { len: 0, cap: 0 } }, Direction { tick: 0, ticks: None, waker: None, wakers: Slab { len: 0, cap: 0 } }], poisoned: false, .. } }, io: Some(UnixStream { fd: FileDesc(OwnedFd { fd: 124 }), local: (unnamed), peer: "/run/user/1000/bus" (pathname) }) }, senders: Mutex { data: {Some(OwnedMatchRule(MatchRule { msg_type: Some(MethodReturn), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName("org.freedesktop.DBus"), reply-serial: 19, body: Signature("s"), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 6, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 1 } }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Signal), sender: Some(BusName::Unique("org.freedesktop.DBus")), interface: Some(InterfaceName("org.freedesktop.DBus")), member: Some(MemberName("NameOwnerChanged")), path_spec: Some(Path(ObjectPath("/org/freedesktop/DBus"))), destination: None, args: [(0, "org.freedesktop.secrets")], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 0, listeners_total: 1 } }, poisoned: false, .. } }, None: Sender { inner: RwLock { data: Inner { queue: [], capacity: 64, receiver_count: 0, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event(<uninitialized>) }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Error), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName("org.freedesktop.DBus"), reply-serial: 19, body: Signature("s"), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 6, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 1 } }, poisoned: false, .. } }} }, already_received_bytes: None, prev_seq: 8, activity_event: Event(<uninitialized>) }
2024-05-11T11:49:13-04:00 [INFO] read_socket; self=SocketReader { socket: Async { source: Source { registration: 124, key: 0, state: Mutex { data: [Direction { tick: 13, ticks: Some((13, 12)), waker: None, wakers: Slab { len: 0, cap: 0 } }, Direction { tick: 0, ticks: None, waker: None, wakers: Slab { len: 0, cap: 0 } }], poisoned: false, .. } }, io: Some(UnixStream { fd: FileDesc(OwnedFd { fd: 124 }), local: (unnamed), peer: "/run/user/1000/bus" (pathname) }) }, senders: Mutex { data: {Some(OwnedMatchRule(MatchRule { msg_type: Some(MethodReturn), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName("org.freedesktop.DBus"), reply-serial: 20, body: Signature(""), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 7, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 1 } }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Signal), sender: Some(BusName::Unique("org.freedesktop.DBus")), interface: Some(InterfaceName("org.freedesktop.DBus")), member: Some(MemberName("NameOwnerChanged")), path_spec: Some(Path(ObjectPath("/org/freedesktop/DBus"))), destination: None, args: [(0, "org.freedesktop.secrets")], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 0, listeners_total: 1 } }, poisoned: false, .. } }, None: Sender { inner: RwLock { data: Inner { queue: [], capacity: 64, receiver_count: 0, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event(<uninitialized>) }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Error), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName("org.freedesktop.DBus"), reply-serial: 20, body: Signature(""), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 7, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 1 } }, poisoned: false, .. } }} }, already_received_bytes: None, prev_seq: 9, activity_event: Event(<uninitialized>) }
2024-05-11T11:49:13-04:00 [INFO] read_socket; self=SocketReader { socket: Async { source: Source { registration: 124, key: 0, state: Mutex { data: [Direction { tick: 14, ticks: Some((14, 13)), waker: None, wakers: Slab { len: 0, cap: 0 } }, Direction { tick: 0, ticks: None, waker: None, wakers: Slab { len: 0, cap: 0 } }], poisoned: false, .. } }, io: Some(UnixStream { fd: FileDesc(OwnedFd { fd: 124 }), local: (unnamed), peer: "/run/user/1000/bus" (pathname) }) }, senders: Mutex { data: {Some(OwnedMatchRule(MatchRule { msg_type: Some(MethodReturn), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName(":1.22"), reply-serial: 21, body: Signature("a{sv}"), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 8, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 1 } }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Signal), sender: Some(BusName::Unique("org.freedesktop.DBus")), interface: Some(InterfaceName("org.freedesktop.DBus")), member: Some(MemberName("NameOwnerChanged")), path_spec: Some(Path(ObjectPath("/org/freedesktop/DBus"))), destination: None, args: [(0, "org.freedesktop.secrets")], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 0, listeners_total: 1 } }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Signal), sender: Some(BusName::WellKnown("org.freedesktop.secrets")), interface: Some(InterfaceName("org.freedesktop.DBus.Properties")), member: Some(MemberName("PropertiesChanged")), path_spec: Some(Path(ObjectPath("/org/freedesktop/secrets/collection/login/6"))), destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 64, receiver_count: 1, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 0, listeners_total: 1 } }, poisoned: false, .. } }, None: Sender { inner: RwLock { data: Inner { queue: [], capacity: 64, receiver_count: 0, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event(<uninitialized>) }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Error), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName(":1.22"), reply-serial: 21, body: Signature("a{sv}"), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 8, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 1 } }, poisoned: false, .. } }} }, already_received_bytes: None, prev_seq: 10, activity_event: Event(<uninitialized>) }
2024-05-11T11:49:13-04:00 [INFO] keep_updated;
2024-05-11T11:49:13-04:00 [INFO] read_socket; self=SocketReader { socket: Async { source: Source { registration: 124, key: 0, state: Mutex { data: [Direction { tick: 15, ticks: Some((15, 14)), waker: None, wakers: Slab { len: 0, cap: 0 } }, Direction { tick: 0, ticks: None, waker: None, wakers: Slab { len: 0, cap: 0 } }], poisoned: false, .. } }, io: Some(UnixStream { fd: FileDesc(OwnedFd { fd: 124 }), local: (unnamed), peer: "/run/user/1000/bus" (pathname) }) }, senders: Mutex { data: {Some(OwnedMatchRule(MatchRule { msg_type: Some(MethodReturn), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 8, receiver_count: 0, inactive_receiver_count: 1, sender_count: 2, head_pos: 10, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 0 } }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Signal), sender: Some(BusName::Unique("org.freedesktop.DBus")), interface: Some(InterfaceName("org.freedesktop.DBus")), member: Some(MemberName("NameOwnerChanged")), path_spec: Some(Path(ObjectPath("/org/freedesktop/DBus"))), destination: None, args: [(0, "org.freedesktop.secrets")], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 0, listeners_total: 1 } }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Signal), sender: Some(BusName::WellKnown("org.freedesktop.secrets")), interface: Some(InterfaceName("org.freedesktop.DBus.Properties")), member: Some(MemberName("PropertiesChanged")), path_spec: Some(Path(ObjectPath("/org/freedesktop/secrets/collection/login/6"))), destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 64, receiver_count: 1, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 0, listeners_total: 1 } }, poisoned: false, .. } }, None: Sender { inner: RwLock { data: Inner { queue: [], capacity: 64, receiver_count: 0, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event(<uninitialized>) }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Error), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 8, receiver_count: 0, inactive_receiver_count: 1, sender_count: 2, head_pos: 10, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 0 } }, poisoned: false, .. } }} }, already_received_bytes: None, prev_seq: 11, activity_event: Event(<uninitialized>) }
2024-05-11T11:49:13-04:00 [INFO] read_socket; self=SocketReader { socket: Async { source: Source { registration: 124, key: 0, state: Mutex { data: [Direction { tick: 16, ticks: Some((16, 15)), waker: None, wakers: Slab { len: 0, cap: 0 } }, Direction { tick: 0, ticks: None, waker: None, wakers: Slab { len: 0, cap: 0 } }], poisoned: false, .. } }, io: Some(UnixStream { fd: FileDesc(OwnedFd { fd: 124 }), local: (unnamed), peer: "/run/user/1000/bus" (pathname) }) }, senders: Mutex { data: {Some(OwnedMatchRule(MatchRule { msg_type: Some(MethodReturn), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName("org.freedesktop.DBus"), reply-serial: 23, body: Signature(""), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 10, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 1 } }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Signal), sender: Some(BusName::Unique("org.freedesktop.DBus")), interface: Some(InterfaceName("org.freedesktop.DBus")), member: Some(MemberName("NameOwnerChanged")), path_spec: Some(Path(ObjectPath("/org/freedesktop/DBus"))), destination: None, args: [(0, "org.freedesktop.secrets")], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 8, receiver_count: 0, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 0 } }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Signal), sender: Some(BusName::WellKnown("org.freedesktop.secrets")), interface: Some(InterfaceName("org.freedesktop.DBus.Properties")), member: Some(MemberName("PropertiesChanged")), path_spec: Some(Path(ObjectPath("/org/freedesktop/secrets/collection/login/6"))), destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 64, receiver_count: 0, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 0 } }, poisoned: false, .. } }, None: Sender { inner: RwLock { data: Inner { queue: [], capacity: 64, receiver_count: 0, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event(<uninitialized>) }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Error), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName("org.freedesktop.DBus"), reply-serial: 23, body: Signature(""), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 10, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 1 } }, poisoned: false, .. } }} }, already_received_bytes: None, prev_seq: 12, activity_event: Event(<uninitialized>) }
2024-05-11T11:49:13-04:00 [INFO] read_socket; self=SocketReader { socket: Async { source: Source { registration: 124, key: 0, state: Mutex { data: [Direction { tick: 17, ticks: Some((17, 16)), waker: None, wakers: Slab { len: 0, cap: 0 } }, Direction { tick: 0, ticks: None, waker: None, wakers: Slab { len: 0, cap: 0 } }], poisoned: false, .. } }, io: Some(UnixStream { fd: FileDesc(OwnedFd { fd: 124 }), local: (unnamed), peer: "/run/user/1000/bus" (pathname) }) }, senders: Mutex { data: {Some(OwnedMatchRule(MatchRule { msg_type: Some(MethodReturn), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName("org.freedesktop.DBus"), reply-serial: 24, body: Signature(""), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 11, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 1 } }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Signal), sender: Some(BusName::Unique("org.freedesktop.DBus")), interface: Some(InterfaceName("org.freedesktop.DBus")), member: Some(MemberName("NameOwnerChanged")), path_spec: Some(Path(ObjectPath("/org/freedesktop/DBus"))), destination: None, args: [(0, "org.freedesktop.secrets")], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [], capacity: 8, receiver_count: 0, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 0 } }, poisoned: false, .. } }, None: Sender { inner: RwLock { data: Inner { queue: [], capacity: 64, receiver_count: 0, inactive_receiver_count: 1, sender_count: 1, head_pos: 0, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event(<uninitialized>) }, poisoned: false, .. } }, Some(OwnedMatchRule(MatchRule { msg_type: Some(Error), sender: None, interface: None, member: None, path_spec: None, destination: None, args: [], arg_paths: [], arg0ns: None })): Sender { inner: RwLock { data: Inner { queue: [(Ok(Msg { type: MethodReturn, sender: UniqueName("org.freedesktop.DBus"), reply-serial: 24, body: Signature(""), fds: [] }), 1)], capacity: 8, receiver_count: 1, inactive_receiver_count: 1, sender_count: 2, head_pos: 11, overflow: false, await_active: false, is_closed: false, send_ops: Event(<uninitialized>), recv_ops: Event { listeners_notified: 18446744073709551615, listeners_total: 1 } }, poisoned: false, .. } }} }, already_received_bytes: None, prev_seq: 13, activity_event: Event(<uninitialized>) }
2024-05-11T11:49:13-04:00 [ERROR] auto-update failed: error:error deserializing release

Caused by:
    expected value at line 1 column 1
2024-05-11T11:49:13-04:00 [INFO] set status on client 159686: Connecting
2024-05-11T11:49:13-04:00 [INFO] Node runtime install_if_needed
2024-05-11T11:49:13-04:00 [ERROR] Error uploading panic to server: 404 Not Found
2024-05-11T11:49:13-04:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-05-11T11:49:14-04:00 [INFO] Initializing default prettier with plugins {}
2024-05-11T11:49:14-04:00 [INFO] starting language servers for HTML: tailwindcss-language-server
2024-05-11T11:49:14-04:00 [INFO] starting language server "tailwindcss-language-server", path: "/home/spite/Projects/my-big-fat-website", id: 1
2024-05-11T11:49:14-04:00 [INFO] Initializing default prettier with plugins {}
2024-05-11T11:49:14-04:00 [INFO] starting language servers for HTML: tailwindcss-language-server
2024-05-11T11:49:14-04:00 [INFO] starting language servers for Go: gopls
2024-05-11T11:49:14-04:00 [INFO] starting language server "gopls", path: "/home/spite/Projects/my-big-fat-website", id: 2
2024-05-11T11:49:14-04:00 [INFO] starting language servers for Go: gopls
2024-05-11T11:49:14-04:00 [INFO] Initializing default prettier with plugins {}
2024-05-11T11:49:14-04:00 [INFO] starting language servers for HTML: tailwindcss-language-server
2024-05-11T11:49:14-04:00 [INFO] Initializing default prettier with plugins {}
2024-05-11T11:49:14-04:00 [INFO] starting language servers for CSS: vscode-css-language-server, tailwindcss-language-server
2024-05-11T11:49:14-04:00 [INFO] starting language server "vscode-css-language-server", path: "/home/spite/Projects/my-big-fat-website", id: 3
2024-05-11T11:49:14-04:00 [INFO] starting language servers for Go: gopls
2024-05-11T11:49:14-04:00 [INFO] starting language servers for Go: gopls
2024-05-11T11:49:14-04:00 [INFO] Initializing default prettier with plugins {}
2024-05-11T11:49:14-04:00 [INFO] starting language servers for HTML: tailwindcss-language-server
2024-05-11T11:49:14-04:00 [INFO] starting language servers for HTML: tailwindcss-language-server
2024-05-11T11:49:14-04:00 [INFO] starting language servers for Go: gopls
2024-05-11T11:49:14-04:00 [INFO] starting language servers for Go: gopls
2024-05-11T11:49:14-04:00 [INFO] starting language servers for HTML: tailwindcss-language-server
2024-05-11T11:49:14-04:00 [INFO] starting language servers for CSS: vscode-css-language-server, tailwindcss-language-server
2024-05-11T11:49:14-04:00 [INFO] starting language servers for Go: gopls
2024-05-11T11:49:14-04:00 [INFO] starting language servers for Go: gopls
2024-05-11T11:49:14-04:00 [INFO] fetching latest version of language server "tailwindcss-language-server"
2024-05-11T11:49:14-04:00 [INFO] fetching latest version of language server "vscode-css-language-server"
2024-05-11T11:49:14-04:00 [INFO] Initializing default prettier with plugins {}
2024-05-11T11:49:14-04:00 [INFO] starting language servers for HTML: tailwindcss-language-server, vscode-html-language-server
2024-05-11T11:49:14-04:00 [INFO] starting language server "vscode-html-language-server", path: "/home/spite/Projects/my-big-fat-website", id: 4
2024-05-11T11:49:14-04:00 [INFO] starting language servers for HTML: tailwindcss-language-server, vscode-html-language-server
2024-05-11T11:49:14-04:00 [INFO] starting language servers for Go: gopls
2024-05-11T11:49:14-04:00 [INFO] starting language servers for Go: gopls
2024-05-11T11:49:14-04:00 [INFO] starting language servers for HTML: tailwindcss-language-server, vscode-html-language-server
2024-05-11T11:49:14-04:00 [INFO] starting language servers for CSS: vscode-css-language-server, tailwindcss-language-server
2024-05-11T11:49:14-04:00 [INFO] starting language servers for Go: gopls
2024-05-11T11:49:14-04:00 [INFO] starting language servers for Go: gopls
2024-05-11T11:49:14-04:00 [INFO] Creating texture 0x560ffd1e3870 of size 1024x1024x1 and format R8Unorm, name 'atlas', handle 0
2024-05-11T11:49:14-04:00 [INFO] Creating buffer 0x560ffcf87890 of size 65536, name 'chunk-0', handle 1
2024-05-11T11:49:14-04:00 [INFO] Creating buffer 0x560ffd1e3da0 of size 4096, name 'chunk-0', handle 2
2024-05-11T11:49:14-04:00 [INFO] Creating buffer 0x560ffd1e4020 of size 26768, name 'chunk-1', handle 3
2024-05-11T11:49:14-04:00 [INFO] Creating buffer 0x560ffd1e40a0 of size 4096, name 'chunk-2', handle 4
2024-05-11T11:49:14-04:00 [INFO] Creating buffer 0x560ffd1e4120 of size 7840, name 'chunk-3', handle 5
2024-05-11T11:49:14-04:00 [INFO] Creating buffer 0x560ffd1e41a0 of size 5600, name 'chunk-4', handle 6
2024-05-11T11:49:14-04:00 [INFO] add connection to peer
2024-05-11T11:49:14-04:00 [INFO] add_connection;
2024-05-11T11:49:14-04:00 [INFO] waiting for server hello
2024-05-11T11:49:14-04:00 [INFO] got server hello
2024-05-11T11:49:14-04:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 403, id: 993260 })
2024-05-11T11:49:14-04:00 [INFO] set status on client 159686: Connected { peer_id: PeerId { owner_id: 403, id: 993260 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-05-11T11:49:14-04:00 [INFO] Creating buffer 0x560ffd0cddd0 of size 4096, name 'chunk-5', handle 7
2024-05-11T11:49:14-04:00 [INFO] Creating buffer 0x560ffd0cde50 of size 26096, name 'chunk-6', handle 8
2024-05-11T11:49:14-04:00 [INFO] found user-installed language server for Go. path: "/home/spite/go/bin/gopls", arguments: ["-mode=stdio"]
2024-05-11T11:49:14-04:00 [INFO] starting language server. binary path: "/home/spite/go/bin/gopls", working directory: "/home/spite/Projects/my-big-fat-website", args: ["-mode=stdio"]
2024-05-11T11:49:14-04:00 [INFO] starting language server. binary path: "/home/spite/.local/share/zed/node/node-v18.15.0-linux-x64/bin/node", working directory: "/", args: ["/home/spite/.local/share/zed/copilot/copilot-v0.5.0/dist/agent.js", "--stdio"]
2024-05-11T11:49:14-04:00 [INFO] Node runtime install_if_needed
2024-05-11T11:49:14-04:00 [INFO] Creating texture 0x560ffd0ce450 of size 1024x1024x1 and format Bgra8Unorm, name 'atlas', handle 9
2024-05-11T11:49:14-04:00 [INFO] Node runtime install_if_needed
2024-05-11T11:49:14-04:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-05-11T15:49:14.587Z] Agent service starting",
  "metadataStr": "[DEBUG] [agent] [2024-05-11T15:49:14.587Z]",
  "extra": [
    "Agent service starting"
  ]
}
2024-05-11T11:49:14-04:00 [INFO] Language server with id 0 sent unhandled notification client/registerCapability:
{
  "registrations": [
    {
      "id": "6d863a08-5eca-46c9-919c-e163164927cb",
      "method": "workspace/didChangeWorkspaceFolders",
      "registerOptions": {}
    }
  ]
}
2024-05-11T11:49:14-04:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-05-11T15:49:14.599Z] Telemetry initialized",
  "metadataStr": "[DEBUG] [agent] [2024-05-11T15:49:14.599Z]",
  "extra": [
    "Telemetry initialized"
  ]
}
2024-05-11T11:49:14-04:00 [INFO] Node runtime install_if_needed
2024-05-11T11:49:15-04:00 [INFO] downloading language server "tailwindcss-language-server"
2024-05-11T11:49:15-04:00 [INFO] Node runtime install_if_needed
2024-05-11T11:49:15-04:00 [INFO] downloading language server "vscode-css-language-server"
2024-05-11T11:49:15-04:00 [INFO] starting language server. binary path: "/home/spite/.local/share/zed/node/node-v18.15.0-linux-x64/bin/node", working directory: "/home/spite/Projects/my-big-fat-website", args: ["/home/spite/.local/share/zed/languages/tailwindcss-language-server/node_modules/.bin/tailwindcss-language-server", "--stdio"]
2024-05-11T11:49:15-04:00 [INFO] Node runtime install_if_needed
2024-05-11T11:49:15-04:00 [INFO] Node runtime install_if_needed
2024-05-11T11:49:15-04:00 [INFO] starting language server. binary path: "/home/spite/.local/share/zed/node/node-v18.15.0-linux-x64/bin/node", working directory: "/home/spite/Projects/my-big-fat-website", args: ["/home/spite/.local/share/zed/languages/vscode-css-language-server/node_modules/vscode-langservers-extracted/bin/vscode-css-language-server", "--stdio"]
2024-05-11T11:49:15-04:00 [INFO] Language server with id 3 sent unhandled notification workspace/diagnostic/refresh:
null
2024-05-11T11:49:15-04:00 [INFO] starting language server. binary path: "/home/spite/.local/share/zed/node/node-v18.15.0-linux-x64/bin/node", working directory: "/home/spite/Projects/my-big-fat-website", args: ["/home/spite/.local/share/zed/extensions/work/html/node_modules/vscode-langservers-extracted/bin/vscode-html-language-server", "--stdio"]
2024-05-11T11:49:16-04:00 [INFO] Language server with id 4 sent unhandled notification workspace/diagnostic/refresh:
null
2024-05-11T11:49:22-04:00 [INFO] Creating buffer 0x560ffd0ce040 of size 23072, name 'chunk-7', handle 10
2024-05-11T11:49:26-04:00 [INFO] Creating buffer 0x560ffd0cee50 of size 89824, name 'chunk-8', handle 11
2024-05-11T11:49:26-04:00 [INFO] Creating buffer 0x560ffd0cec50 of size 101248, name 'chunk-9', handle 12
2024-05-11T11:49:34-04:00 [INFO] Creating texture 0x560ffd0cf370 of size 1024x1024x1 and format R16Float, name 'atlas', handle 13
2024-05-11T11:49:58-04:00 [INFO] Destroying texture 0x560ffd1e3870, handle 0
2024-05-11T11:49:58-04:00 [INFO] Destroying texture 0x560ffd0ce450, handle 9
2024-05-11T11:49:58-04:00 [INFO] Destroying texture 0x560ffd0cf370, handle 13
2024-05-11T11:49:58-04:00 [INFO] Destroying buffer 0x560ffcf87890, handle 1
2024-05-11T11:49:58-04:00 [INFO] Destroying buffer 0x560ffd1e41a0, handle 6
2024-05-11T11:49:58-04:00 [INFO] Destroying buffer 0x560ffd0cde50, handle 8
2024-05-11T11:49:58-04:00 [INFO] Destroying buffer 0x560ffd0cddd0, handle 7
2024-05-11T11:49:58-04:00 [INFO] Destroying buffer 0x560ffd0ce040, handle 10
2024-05-11T11:49:58-04:00 [INFO] Destroying buffer 0x560ffd1e4120, handle 5
2024-05-11T11:49:58-04:00 [INFO] Destroying buffer 0x560ffd1e3da0, handle 2
2024-05-11T11:49:58-04:00 [INFO] Destroying buffer 0x560ffd0cee50, handle 11
2024-05-11T11:49:58-04:00 [INFO] Destroying buffer 0x560ffd1e40a0, handle 4
2024-05-11T11:49:58-04:00 [INFO] Destroying buffer 0x560ffd1e4020, handle 3
2024-05-11T11:49:58-04:00 [INFO] Destroying buffer 0x560ffd0cec50, handle 12
2024-05-11T11:49:58-04:00 [ERROR] crates/lsp/src/lsp.rs:768: oneshot canceled
2024-05-11T11:49:58-04:00 [ERROR] crates/lsp/src/lsp.rs:768: oneshot canceled
2024-05-11T11:49:58-04:00 [ERROR] crates/lsp/src/lsp.rs:768: oneshot canceled
2024-05-11T11:49:58-04:00 [ERROR] crates/lsp/src/lsp.rs:768: oneshot canceled

--- a/crates/gpui/src/platform/cosmic_text/text_system.rs
+++ b/crates/gpui/src/platform/cosmic_text/text_system.rs
@@ -314,7 +314,7 @@ impl CosmicTextSystemState {
             let bitmap_size = glyph_bounds.size;
             let font = &self.loaded_fonts_store[params.font_id.0];
             let font_system = &mut self.font_system;
-            let image = self
+            let mut image = self
                 .swash_cache
                 .get_image(
                     font_system,
@@ -329,6 +329,13 @@ impl CosmicTextSystemState {
                 )
                 .clone()
                 .unwrap();
+
+            if params.is_emoji {
+                // Convert from RGBA to BGRA.
+                for pixel in image.data.chunks_exact_mut(4) {
+                    pixel.swap(0, 2);
+                }
+            }
 
             Ok((bitmap_size, image.data))
         }

--- a/crates/gpui/src/platform/cosmic_text/text_system.rs
+++ b/crates/gpui/src/platform/cosmic_text/text_system.rs
@@ -394,13 +394,20 @@ impl CosmicTextSystemState {
         for glyph in &layout.glyphs {
             let font_id = glyph.font_id;
             let font_id = self.font_id_for_cosmic_id(font_id);
+            let is_emoji = self.is_emoji(font_id);
             let mut glyphs = SmallVec::new();
+
+            // HACK: Prevent crash caused by variation selectors.
+            if glyph.glyph_id == 3 && is_emoji {
+                continue;
+            }
+
             // todo(linux) this is definitely wrong, each glyph in glyphs from cosmic-text is a cluster with one glyph, ShapedRun takes a run of glyphs with the same font and direction
             glyphs.push(ShapedGlyph {
                 id: GlyphId(glyph.glyph_id as u32),
                 position: point((glyph.x).into(), glyph.y.into()),
                 index: glyph.start,
-                is_emoji: self.is_emoji(font_id),
+                is_emoji,
             });
 
             runs.push(crate::ShapedRun { font_id, glyphs });

